### PR TITLE
Add test method for DishViewCommand

### PR DESCRIPTION
**Pull request by Zeno-Zr**
Restructuring the following features allows us to fulfil the features relating to menu commands
* Add test method for DishViewCommand method in DinerDirectorTest class
* Refactor code for DishAddCommand method in Parser class

### CODE REVIEW

File: src/main/java/utils/Parser.java [hunk 1]
```diff
@@ -197,37 +200,54 @@ private Command prepareDeleteDishCommand(String[] userInputSplit) {
...
+            return new IncorrectCommand();
+        }
+
         String name;
         int price;
         ArrayList<String> ingredients;
 
         try {
-            System.out.println("Name of Dish?");
             Scanner userInput = new Scanner(System.in);
```
Comment (Stella1585) [inline src/main/java/utils/Parser.java:233]: maybe make use of the ui class

File: src/main/java/utils/Parser.java [hunk 2]
```diff
@@ -197,37 +200,54 @@ private Command prepareDeleteDishCommand(String[] userInputSplit) {
...
+            System.out.println("Name of Dish?");
```
Comment (Stella1585) [inline src/main/java/utils/Parser.java:234]: maybe you can follow a similar format to our other commands, n/dish_name p/price. then for ingredients you can do what you are doing now

File: src/main/java/utils/Parser.java [hunk 3]
```diff
@@ -183,6 +183,9 @@ private Command prepareDeleteDeadlineCommand(String description) {
     private Command prepareDeleteDishCommand(String[] userInputSplit) {
         int indexToRemove = 0;
         try {
+            if (userInputSplit.length <= 1) {
+                throw new DinerDirectorException(Messages.MESSAGE_EMPTY_INDEX_COMMAND);
```
Comment (darrenangwx) [inline src/main/java/utils/Parser.java:187]: Good that you are handling exceptions

**Review by darrenangwx — APPROVED** [review]

File: src/main/java/utils/Parser.java [hunk 4]
```diff
@@ -197,37 +200,54 @@ private Command prepareDeleteDishCommand(String[] userInputSplit) {
...
             if (price < 0) {
                 throw new DinerDirectorException(Messages.MESSAGE_NEGATIVE_PRICE_COMMAND);
             }
 
             System.out.println("List of ingredients? (Separate it by spaces)");
-            userInput = new Scanner(System.in);
+            if (!userInput.hasNext()) {
+                throw new DinerDirectorException(Messages.MESSAGE_INGREDIENT_LIST_CANNOT_BE_EMPTY);
+            }
             String[] userInputSplit = userInput.nextLine().split(" ");
```
Comment (Stella1585) [inline src/main/java/utils/Parser.java:251]: are you sure all ingredients can be split with " ", like baking powder or smt, maybe ask them to split with something else. like i/name1 i/name2 i/name3 than split with " i/" will be better?

also i think u need to check with Darren regarding the help guide cos urs is diff from our user guide commands.

File: src/test/java/dinerdirector/DinerDirectorTest.java [hunk 5]
```diff
@@ -66,4 +72,105 @@ void runCommandLoopUntilExit_userInput_deadlineCommand() {
...
+        final String TEST_CASE_DECIMAL_NUMBER = "apple pie 2nd edition" + System.lineSeparator() +
+                "3.93";
+        final String TEST_CASE_NEGATIVE_INTEGER = "apple pie 2nd edition" + System.lineSeparator() +
+                "-1";
+        final String TEST_CASE_EMPTY_INGREDIENT_LIST = "apple pie 2nd edition" + System.lineSeparator() +
+                "321" + System.lineSeparator() +
+                "";
+        final String TEST_CASE_CORRECT_INPUT = "apple pie 2nd edition" + System.lineSeparator() +
+                "321" + System.lineSeparator() +
+                "apple flour water butter";
```
Comment (douph810975) [inline src/test/java/dinerdirector/DinerDirectorTest.java:107]: It's good that you add lots of test cases

File: src/test/java/dinerdirector/DinerDirectorTest.java [hunk 6]
```diff
@@ -66,4 +72,105 @@ void runCommandLoopUntilExit_userInput_deadlineCommand() {
...
+
+        ArrayList<String> listOfCommands = new ArrayList<>();
+
+        final String SETUP_ADD_DISH = "add_dish";
+
+        final String TEST_CASE_NO_INDEX = "delete_dish";
+        final String TEST_CASE_NEGATIVE_INDEX = "delete_dish -1";
+        final String TEST_CASE_ZERO_INDEX = "delete_dish 0";
+        final String TEST_CASE_OUT_OF_BOUNDS_INDEX = "delete_dish 2";
+        final String TEST_CASE_VALID_INDEX = "delete_dish 1";
```
Comment (douph810975) [inline src/test/java/dinerdirector/DinerDirectorTest.java:153]: I like your efforts to try different scenarios.

**Review by douph810975 — COMMENTED** [review]
Good and comprehensive test file.

**Pull request merged**